pelican_bib/bib.py

```diff
@@ -160,6 +201,159 @@ def add_publications(generator):
+          :pybtex_style_args: PYBTEX_STYLE_ARGS (optional)
+          :sorting_style: PYBTEX_SORTING_STYLE (optional)
+          :abbreviate_names: PYBTEX_ABBREVIATE_NAMES (optional)
+          :name_style: PYBTEX_NAME_STYLE (optional)
+
+    e.g.::
+
+        .. bibliography:: osm.bib
+          :template: publications_unsrt
+          :options: {'groupby_value': year}
+          :class: my-bib
+          :pybtex_style_args: {'sorting_style': 'author_year_title',
+                               'abbreviate_names': False,
+                               'name_style': 'lastfirst' }
+          :sorting_style: author_year_title
+          :abbreviate_names: False
+          :name_style: lastfirst
+    """
+    required_arguments = 0
+    optional_arguments = 100
+    final_argument_whitespace = False
+    has_content = True
+
+    generator = None
+
+    def set_generator(generator):
+        Bibliography.generator = generator
+
+    def boolean(argument):
+        if argument == 'True': return True
+        if argument == 'False': return False
+        raise ValueError('must be boolean')
+    option_spec = {
+        'template': directives.unchanged,
+        'options': directives.unchanged,
+        'class': directives.class_option,
+        'filter_tag': directives.unchanged,
+        'pybtex_style_args': directives.unchanged,
+        'sorting_style': directives.unchanged,
+        'abbreviate_names': boolean,
+        'name_style': directives.unchanged
+    }
+
+    def run(self):
+        refs_files = self.arguments
+        refs_string = '\n'.join(self.content)
+        template_name = self.generator.settings.get(
+            'PUBLICATIONS_DEFAULT_TEMPLATE', 'bibliography'
+        )
+        template_options = {}
+        classes = ['bibliography']
+        filter_tag = None
+        pybtex_style_args = {}
+
+        # fetch arguments
+        if refs_files:
+            classes += (basename(file) for file in refs_files)
+        if 'template' in self.options:
+            template_name = self.options['template']
+        if 'options' in self.options:
+            template_options = literal_eval(self.options['options'])
+        if 'class' in self.options:
+            classes = self.options['class']
+        if 'filter_tag' in self.options:
+            filter_tag = self.options['filter_tag']
+        if 'pybtex_style_args' in self.options:
+            pybtex_style_args.update(literal_eval(self.options['pybtex_style_args']))
+        if 'sorting_style' in self.options:
+            pybtex_style_args['sorting_style'] = self.options[
+                'sorting_style'
+            ]
+        if 'abbreviate_names' in self.options:
+            pybtex_style_args['abbreviate_names'] = self.options['abbreviate_names']
+        if 'name_style' in self.options:
+            pybtex_style_args['name_style'] = self.options['name_style']
+
+        # BibTeX input is a bib file or the directives content
+        if not self.content and not refs_files:
```

Comment:
Regarding the custom pybtex styles introduced by @eginhard ... it would nice to have them integrated in the directive too. If I set `PUBLICATIONS_CUSTOM_STYLE = True` globally, I can easily pass arguments to the custom style using the directive's `pybtex_style_args` option. It might be nice to be able to switch the use of custom styles on and off maybe? 

As an example: I might want to underline my name in the list of personal publications. But when I speak about a project in general and add a series of publications, that is probably not of interest. What do you think? (Similar to the suggestion to introduce `use_global_src`, this can be also treated as a feature request later).

Reply:
This sounds like an great feature request. :-)